src/main/kotlin/org/rust/ide/inspections/RsSelfConventionInspection.kt

```diff
@@ -19,12 +19,12 @@ class RsSelfConventionInspection : RsLocalInspectionTool() {
     override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) =
         object : RsVisitor() {
             override fun visitFunction(m: RsFunction) {
-                // Should this handle traits as well perhaps?
-                if (m.owner !is RsFunctionOwner.Impl) return
+                val owner = m.owner
+                if (!(owner.isInherentImpl || owner.isTrait)) return
 
                 val convention = SELF_CONVENTIONS.find { m.identifier.text.startsWith(it.prefix) } ?: return
                 if (m.selfType in convention.selfTypes) return
-                if (m.selfType == SelfType.SELF && m.isOwnerCopyable()) return
+                if (owner.isInherentImpl && m.selfType == SelfType.SELF && m.isOwnerCopyable()) return
```

Comment:
Why do we need `owner.isInherentImpl` here?

Reply:
isOwnerCopy works only on Impl